File: root/Assets/Scripts/Movement.js

#pragma strict
private var movement : Vector3 = Vector3.zero;
private var controller : CharacterController;

var bullet : GameObject;
var fireRate = 0.5;
var nextFire = 0;
var lives : int;
var lifetime = 2.0;





function Start () 
{
    controller = GetComponent(CharacterController);
}


function OnTriggerEnter(col : Collider)
    {
        if(col.gameObject.tag == "Enemy")
        {
            GameObject.Destroy(col.gameObject);
            GameObject.Destroy(this.gameObject);
            Application.LoadLevel("map_layout");
        }
    }


    /*function OnDestroy()
    {
        transform.parent.gameObject.AddComponent<GameLose>();
    }*/

    function Update () 
    {
        movement = Vector3(Input.GetAxis("Horizontal"), 0, 0);
	
        movement = transform.TransformDirection(movement);
	
        movement *= 7;
	
        controller.Move(movement * Time.deltaTime);

      //  print(GameObject.FindGameObjectsWithTag("Enemy").Length);
        if(GameObject.FindGameObjectsWithTag("Enemy").Length==0)
        {
            Application.LoadLevel("map_layout");
        }

        

    //print(GameObject.FindGameObjectsWithTag("Enemy"));
    /*if(GameObject.FindGameObjectsWithTag("Enemy") == 0);
    {
        Application.LoadLevel(0);
    }*/

   


    if (Input.GetKeyUp("up") && Time.time > nextFire)
        {
            var spawn_bullet = Instantiate (bullet, transform.position, Quaternion.identity);
            spawn_bullet.GetComponent.<Rigidbody>().AddForce(Vector3.up * 115);
            nextFire = Time.time + fireRate;
           // var clone = Instantiate (bullet, transform.position, transform.rotation);

    }
  
    
}

File: root/Assets/Scripts/Movement.cs
using UnityEngine;
using System.Collections;
//using Assets.Scripts;

public class Movement : MonoBehaviour {
    private CharacterController controller;
    public Vector3 movement;
    public GameObject bullet;
    public Rigidbody bulletPrefab;
    public float cooldown = 30.0f;
    private float nextFire = 30.0f;
   
    
    
	// Use this for initialization
	void Start () {
        
        ShootBullets();
	}
	
	// Update is called once per frame
	void Update () {
        movement = new Vector3(Input.GetAxis("Horizontal"), 0, 0);
        movement = transform.TransformDirection(movement);
        movement *= 2;
        controller.Move(movement * Time.deltaTime);
	}

    void ShootBullets()
    {
        
          if (Input.GetKey("space") && Time.time >= nextFire)
       {
           nextFire += cooldown;
           Instantiate(bullet, transform.position, transform.rotation);
           //int spawn_bullet = Instantitate(bullet, Transform.position, Quaternion.identity);
           //spawn_bullet.GetComponent<Rigidbody>().AddForce(Vector3.up * 50);
       }
    }

   /* if(Input.GetKey("space"))
{
   int  spawn_bullet = Instantiate(bullet,Transform.position,Quaternion.identity);
    spawn_bullet.GetComponent.<Rigidbody>().AddForce(Vector3.up * 50);
}*/

}
